src/main/java/org/openlmis/stockmanagement/controller/StockCardController.java

```diff
@@ -398,5 +452,24 @@ private void removeEmptyLotsFromStockCard(StockCard stockCard)
         stockCard.setLotsOnHand(nonEmptyLots);
     }
 
+    private List<Right> getRightsForUserFacilityAndProductCode(Long userId, Long facilityId, String productCode)
+    {
+        Facility facility = facilityRepository.getById(facilityId);
+
+        // Get programs by product code, through programProducts
```

Comment:
Since this deals with rights, it might be best to put it in the RoleRightsService then.